### PR TITLE
added action mailer functionality

### PR DESCRIPTION
**Pull request by Tramchandani**

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -10,6 +10,7 @@ def create
     if @user.save 
       session[:user_id] = @user.id
       flash[:notice] = "signup successful"
+      UserMailer.with(user: @user).welcome_email.deliver_later
```
Comment (kalpana19sharma) [inline app/controllers/users_controller.rb]: Let's add this functionality in model. So that if any user is created from console or from any other controller you don't need to remember this email functionality.

File: config/environments/development.rb [hunk 2]
```diff
@@ -34,7 +34,7 @@
   config.active_storage.service = :local
 
   # Don't care if the mailer can't send.
-  config.action_mailer.raise_delivery_errors = false
+  #config.action_mailer.raise_delivery_errors = false
```
Comment (kalpana19sharma) [inline config/environments/development.rb:37]: Why did you remove this?

File: config/environments/development.rb [hunk 3]
```diff
@@ -73,4 +73,16 @@
 
   # Uncomment if you wish to allow Action Cable access from any origin.
   # config.action_cable.disable_request_forgery_protection = true
+  config.action_mailer.perform_deliveries = true
+  config.action_mailer.raise_delivery_errors = true
+  config.action_mailer.default_url_options = { :host => 'localhost:3000', protocol: 'http' }
```
Comment (kalpana19sharma) [inline config/environments/development.rb]: Don't use arrow syntax

File: config/environments/production.rb [hunk 4]
```diff
@@ -117,4 +117,16 @@
   # config.active_record.database_selector = { delay: 2.seconds }
   # config.active_record.database_resolver = ActiveRecord::Middleware::DatabaseSelector::Resolver
   # config.active_record.database_resolver_context = ActiveRecord::Middleware::DatabaseSelector::Resolver::Session
+  config.action_mailer.perform_deliveries = true
+  config.action_mailer.raise_delivery_errors = true
+  config.action_mailer.default_url_options = { :host => 'localhost:3000', protocol: 'http' }
```
Comment (kalpana19sharma) [inline config/environments/production.rb]: Is it okay to use localhost in production as well?

**Review by kalpana19sharma — CHANGES_REQUESTED** [review]

File: config/environments/development.rb (hunk 2 above)
Comment (Tramchandani) [reply config/environments/development.rb:37]: ma'am I have set it as true below in the same file and later I noticed this was already present as the default setting so I commented it.

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -10,6 +10,7 @@ def create
     if @user.save 
       session[:user_id] = @user.id
       flash[:notice] = "signup successful"
+      @user.mail
```
Comment (kalpana19sharma) [inline app/controllers/users_controller.rb]: Can we shift it in a after create callback in model?

**Review by kalpana19sharma — CHANGES_REQUESTED** [review]

**Review by kalpana19sharma — APPROVED** [review]
